Jenkinsfile

```diff
@@ -1,52 +1,59 @@
 elifePipeline {
     node('containers-jenkins-plugin') {
         def commit
-        def grobidTag
-        def fullImageTag
+        def allGrobidTags
 
         stage 'Checkout', {
             checkout scm
             commit = elifeGitRevision()
-            grobidTag = sh(
-                script: 'bash -c "source .env && echo \\$GROBID_TAG"',
+            allGrobidTags = sh(
+                script: 'bash -c "source .env && echo \\$ALL_GROBID_TAGS"',
                 returnStdout: true
-            ).trim()
-            echo "GROBID_TAG: ${grobidTag}"
-            assert grobidTag != ''
-            fullImageTag = "${grobidTag}-${commit}"
-            echo "Full image tag: ${fullImageTag}"
+            ).trim().split(':')
+            echo "allGrobidTags: ${allGrobidTags}"
+            assert allGrobidTags != []
         }
 
         stage 'Build and run tests', {
             try {
-                sh "IMAGE_TAG=${fullImageTag} REVISION=${commit} make ci-build-and-test"
+                parallel(allGrobidTags.inject([:]) { m, grobidTag ->
+                    m["Build and run tests (${grobidTag})"] = {
+                        def fullImageTag = "${grobidTag}-${commit}"
+                        sh "GROBID_TAG=${grobidTag} IMAGE_TAG=${fullImageTag} REVISION=${commit} make ci-build-and-test"
+
+                        echo "Checking GROBID label..."
+                        def image = DockerImage.elifesciences(this, 'sciencebeam-trainer-grobid', fullImageTag)
+                        echo "Reading GROBID label of image: ${image.toString()}"
+                        def actualGrobidTag = sh(
+                            script: "./ci/docker-read-local-label.sh ${image.toString()} org.elifesciences.dependencies.grobid",
+                            returnStdout: true
+                        ).trim()
+                        echo "GROBID label: ${actualGrobidTag} (expected: ${grobidTag})"
+                        assert actualGrobidTag == grobidTag
+                    }
+                    return m
+                })
             } finally {
-                sh "IMAGE_TAG=${fullImageTag} REVISION=${commit} make ci-clean"
+                sh "make ci-clean"
```

Comment:
due to Jenkins pipeline visualization, it would be best for the `parallel` block to be the only one inside a `stage`. Not sure if `make ci-clean` can be pushed in it however, as the concurrent executions of it may clash.

Reply:
(nice to have, not mandatory)

Reply:
this is basically `docker-compose down -v` which I believe you also added at that level in [sciencebeam-utils](https://github.com/elifesciences/sciencebeam-utils/blob/develop/Jenkinsfile#L54) or am I misreading that?

Reply:
Yes, if it is `docker-compose down` it must be executed only once. Could add a lock to enforce it... but not bother.